orttraining/orttraining/python/training/orttrainer.py

```diff
@@ -843,3 +854,306 @@ def _update_onnx_model_initializers(self, state_tensors):
+                    {
+                        model_weight_name:
+                        {
+                            "original_dim": original_dimension_array
+                        }
+                    }
+                }
+            - when pytorch_format is true
+                {
+                    model_weight_name: model_weight_tensor
+                }
+
+        """
+        if not self._training_session:
+            warnings.warn("ONNX Runtime training session is not initialized yet. " \
+                          "Please run train_step or eval_step at least once before calling ORTTrainer.state_dict().", \
+                          UserWarning)
+            return self._load_state_dict.args[0] if self._load_state_dict else {}
+
+        states = {}
+
+        # load training session model states into the states dictionary
+        self._extract_model_states(states)
+        if pytorch_format:
+            # if pytorch_format is true, return a flat dictionary with only model states
+            # which is compatible with a PyTorch model
+            return states['model']['fp32']
+
+        # load training session optimizer states into the states dictionary
+        self._extract_optimizer_states(states)
+
+        # extract the relevant training configuration from the trainer and load them into the states dictionary
+        self._extract_trainer_options(states)
+
+        # add partition information in case of a distributed run
+        if self.options.distributed.deepspeed_zero_optimization.stage > 0:
+            partition_info = 'partition_info'
+            states[partition_info] = self._training_session.get_partition_info_map()
+
+        return states
+
+    def _load_model_states(self, training_session_state_dict, state_dict, strict):
+        """Load the model states onto the training session state dictionary"""
+
+        model = 'model'
+
+        # collect all initializer names from the current onnx graph
+        initializer_names = {node.name for node in self._onnx_model.graph.initializer}
+
+        # loaded_initializers dict will be loaded with all the model states from the state dictionary
+        # that are found in the initializer_names dictionary
+        loaded_initializers = {}
+
+        # create an entry for the model in the training session state dictionary
+        if model not in training_session_state_dict:
+            training_session_state_dict[model] = {}
+
+        # copy over model states from the input state dict onto the training session state dict
+        # and copy over model states from the input state dict onto the onnx model
+        for precision, precision_states in state_dict[model].items():
+            if precision not in training_session_state_dict[model]:
+                training_session_state_dict[model][precision] = {}
+
+            for state_key, state_value in precision_states.items():
+                state_value_np = torch.tensor(state_value).numpy()
```

Comment:
why not numpy.array(state_value) ?

Reply:
yeah, I can do that. thanks, will update.